fastestimator/trace/metric/bleu_score.py

```diff
@@ -0,0 +1,121 @@
+       the Machine Translation output. The closer to 1 the test sentences score, the more overlap there is with their
+       human reference translations and thus, the better the system is deemed to be. The MT output would score 1 only
+       if it is identical to the reference human translation. But even two competent human translations of the exact
+       same material may only score in the 0.6 or 0.7 range as they are likely to use different vocabulary and phrasing.
+       We should be wary of very high BLEU scores (in excess of 0.7) as it is probably measuring improperly or overfitting.
+
+       The default BLEU calculates a score for up to 4-grams using uniform weights (this is called BLEU-4). To evaluate
+       your translations with lower order ngrams, use customized "n_gram". E.g. when accounting for up to 2-grams
+       with uniform weights (this is called BLEU-2) use n_gram=2.
+
+    Args:
+        true_key: Name of the key that corresponds to ground truth in the batch dictionary.
+        pred_key: Name of the key that corresponds to predicted score in the batch dictionary.
+        mode: What mode(s) to execute this Trace in. For example, "train", "eval", "test", or "infer". To execute
+            regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
+            like "!infer" or "!train".
+        ds_id: What dataset id(s) to execute this Trace in. To execute regardless of ds_id, pass None. To execute in all
+            ds_ids except for a particular one, you can pass an argument like "!ds1".
+        output_name: Name of the key to store back to the state.
+        n_gram: Number of grams which needs to be used to calculate bleu score(default: 4)
+        per_ds: Whether to automatically compute this metric individually for every ds_id it runs on, in addition to
+            computing an aggregate across all ds_ids on which it runs. This is automatically False if `output_name`
+            contains a "|" character.
+
+    """
+    def __init__(self,
+                 true_key: str,
+                 pred_key: str,
+                 mode: Union[None, str, Iterable[str]] = ("eval", "test"),
+                 ds_id: Union[None, str, Iterable[str]] = None,
+                 output_name: str = 'bleu_score',
+                 n_gram: int = 4,
+                 per_ds: bool = True) -> None:
+        super().__init__(inputs=(true_key, pred_key), outputs=output_name, mode=mode, ds_id=ds_id)
+        #BleuScore trace adds 1 field bleu_score to state dict
+        self.n_gram = n_gram
+        self.weights = self.get_output_weights()
+        self.sentence_level_bleu_scores = []
+        self.per_ds = per_ds
+
+    @property
+    def true_key(self) -> str:
+        return self.inputs[0]
+
+    @property
+    def pred_key(self) -> str:
+        return self.inputs[1]
+
+    def get_output_weights(self):
+        if self.n_gram == 1:
```

Comment:
is there a reason that it can only go up to 4? why not just assert that it's >0 and then do: return (1/self.n_gram, ) * self.n_gram